src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTriggerTest.java

```diff
@@ -1733,6 +1733,87 @@ public void testDependencyValidationThreeProjectsInvolved() {
                 descriptor.doCheckDependencyJobsNames(downstreamProject, "MockedUpstreamProject").kind);
     }
 
+    /**
+     * Tests that dynamic project configurations do not miss an event from the time
+     * at which the trigger was started until the time at which the URL was fetched.
+     * @throws Exception on failure
+     */
+    @PrepareForTest({
+            GerritTrigger.class,
+            AbstractProject.class,
+            ToGerritRunListener.class,
+            PluginImpl.class,
+            Hudson.class,
+            Jenkins.class,
+            DependencyQueueTaskDispatcher.class,
+            EventListener.class })
+    @Test
+    public void testDynamicTriggerConfigurationTimeGap() throws Exception {
+        AbstractProject project = PowerMockito.mock(AbstractProject.class);
+        when(project.getFullName()).thenReturn("MockedProject");
+        when(project.isBuildable()).thenReturn(true);
+
+        Queue queue = mockConfig(project);
+
+        PowerMockito.mockStatic(ToGerritRunListener.class);
+        ToGerritRunListener listener = PowerMockito.mock(ToGerritRunListener.class);
+        PowerMockito.when(ToGerritRunListener.getInstance()).thenReturn(listener);
+
+        // Set up a temporary file to use as our dynamic configuration URL.
+        java.nio.file.Path temporaryConfigFile = java.nio.file.Files.createTempFile("GerritTriggerTest", null);
+        java.nio.file.Files.write(temporaryConfigFile, "p=my-project\nb^**".getBytes());
+        System.out.println("Temporary file: " + temporaryConfigFile.toString());
+
+        GerritTrigger trigger = new GerritTrigger(null);
+        trigger.setDynamicTriggerConfiguration(true);
+        trigger.setTriggerConfigURL("file://" + temporaryConfigFile.toString());
+
+        // Set up the job within the trigger.
+        Whitebox.setInternalState(trigger, "job", project);
+
+        // We need to make sure that whenever anyone asks for the trigger for any job
+        // that it returns this one.
+        PowerMockito.mockStatic(GerritTrigger.class);
+        // TODO: Why doesn't this work?
+        // TODO: PowerMockito.stub(PowerMockito.method(GerritTrigger.class, "getTrigger")).toReturn(trigger);
+        when(GerritTrigger.getTrigger(any(Job.class))).thenReturn(trigger);
```

Comment:
It might not work because you perhaps need to use PowerMockito's when method?

Reply:
I detect a hint of sarcasm there.  From what I had read on `PowerMockito` last year, it was my understanding that the `stub` method was the proper way to handle a static method on a class, but I have no interest in revisiting this since `when` works.  It is what it is.

Comments have been removed.